tikv/kv.go

```diff
@@ -646,13 +637,17 @@ func (s *KVStore) updateMinSafeTS(txnScope string, storeIDs []uint64) {
 		ok, safeTS := s.getSafeTS(store)
 		if ok {
 			// safeTS is guaranteed to be less than math.MaxUint64 (by setSafeTS and its callers)
-			if safeTS < minSafeTS {
+			if safeTS != 0 && safeTS < minSafeTS {
 				minSafeTS = safeTS
 			}
 		} else {
 			minSafeTS = 0
 		}
 	}
+	// if minSafeTS is still math.MaxUint64, that means all store safe ts are 0, then we set minSafeTS to 0.
+	if minSafeTS == math.MaxUint64 {
+		minSafeTS = 0
```

Comment:
Or could we set it to something like `safepoint` ts ?

Reply:
Sound reasonable, let me think about if it would break some constraints.